pontoon/base/views.py

```diff
@@ -988,50 +993,75 @@ def transvision(request):
         return HttpResponse("error")
 
 
+def get_download_content(slug, code, relative_path):
+    """
+    Get content for the file to be downloaded.
+
+    :param str slug: Project slug.
+    :param str code: Locale code.
+    :param str relative_path: Resource path.
+    """
+    # Clone or update project repository
+    project = get_object_or_404(Project, slug=slug)
+    project.repositories.first().pull()
+
+    # Get source file path
+    locale = get_object_or_404(Locale, code__iexact=code)
```

Comment:
This should be above the point where we pull files, as we don't want to trigger a pull only to abort with a 404.